### PR TITLE
Add mutex support

### PR DESCRIPTION
**Pull request by rbudai98**
PR:
- mutex header added
- mutex (empty) implementation for no_os

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
the two files added in this PR are interdependent, therefore they should go in the same commit (squash commits)

**Comment by rbudai98** [comment]
v2:
- squashed commits

File: util/no_os_mutex.c [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * @return None.
+ */
+__attribute__((weak)) inline void no_os_mutex_lock(void *ptr) {}
+
+/**
+ * @brief Unlock mutex.
+ * @param ptr - Pointer toward the mutex.
+ * @return None.
+ */
+__attribute((weak)) inline void no_os_mutex_lock(void *ptr) {}
```
Comment (amiclaus) [inline util/no_os_mutex.c:61]: `no_os_mutex_unlock`?

**Review by amiclaus — COMMENTED** [review]
is this still WIP? If not, I'd convert it back to PR.

File: include/no_os_mutex.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef _NO_OS_MUTEX_H_
+#define _NO_OS_MUTEX_H_
+
+/* Initialize mutex */
+void no_os_mutex_init(void *ptr);
```
[the remark this answers is not in the document]
Comment (rbudai98) [reply include/no_os_mutex.h:43]: like:
void no_os_mutex_init(double *mutex)?

[the remark this answers is not in the document]
Comment (buha) [reply include/no_os_mutex.h:43]: `void no_os_mutex_init(void **mutex)?`

[the remark this answers is not in the document]
Comment (rbudai98) [reply include/no_os_mutex.h:43]: Okay, got it :)

**Comment by rbudai98** [comment]
v3:
- ptr modified to mutex
- mutex changed to double pointer in init function

**Review by buha — APPROVED** [review]

**Pull request merged**